coreclient/src/conversations/messages/persistence.rs

```diff
@@ -51,37 +56,37 @@ enum VersionedMessageInputs {
     CurrentVersion(Vec<u8>, MessageInputs),
 }
 
-impl Message {
-    // For future message types, the additional inputs to this function might
-    // have to be adjusted.
-    fn from_versioned_message(
-        versioned_message: VersionedMessageInputs,
-    ) -> Result<Self, phnxtypes::codec::Error> {
-        match versioned_message {
-            VersionedMessageInputs::CurrentVersion(message_bytes, inputs) => match inputs {
-                MessageInputs::System => {
-                    let event_message = PhnxCodec::from_slice(&message_bytes)?;
-                    Ok(Message::Event(event_message))
-                }
-                MessageInputs::User(sender, sent) => {
-                    let content = PhnxCodec::from_slice(&message_bytes)?;
-                    let content_message = ContentMessage {
-                        sender,
-                        sent,
-                        content,
-                    };
-                    Ok(Message::Content(Box::new(content_message)))
-                }
-            },
+impl VersionedMessage {
+    fn to_event_message(self) -> Result<EventMessage, phnxtypes::codec::Error> {
+        match self.version {
+            1 => PhnxCodec::from_slice::<EventMessage>(&self.content),
+            _ => unimplemented!(),
```

Comment:
My code still uses some panicking macros. I should probably remove them before merging